src/compiler/checker.ts

```diff
@@ -5106,7 +5094,9 @@ namespace ts {
         }
 
         function combineTypeMappers(mapper1: TypeMapper, mapper2: TypeMapper): TypeMapper {
-            return t => instantiateType(mapper1(t), mapper2);
+            const mapper: TypeMapper = t => instantiateType(mapper1(t), mapper2);
+            mapper.mappedTypes = mapper1.mappedTypes;

```

Comment:
No, from an outside point of view, a combined mapper only re-maps the type parameters of `mapper1` (the one in front). The fact that more mapping happens below is really irrelevant. Either way, with the current analysis we should never actually see a combined mapper because that only happens when we're instantiating already instantiated types (and we don't analyse those now).